pkg/testutil/fakes_test.go

```diff
@@ -390,6 +392,86 @@ func TestSimpleServiceLister(t *testing.T) {
 	assert.Equal(t, "test", service.Namespace)
 }
 
+// TestSimpleClusterRoleLister tests the functionality of simpleClusterRoleLister
+// GIVEN a cluster which has no existing cluster roles
+//  WHEN I create two cluster roles and a simpleClusterRoleLister
+//  THEN the lister should list exactly two cluster roles given an everything selector
+//   AND the lister should list exactly one specific cluster role given a label selector
+//   AND the lister should get a specific cluster role when requested by name
+func TestSimpleClusterRoleLister(t *testing.T) {
+	// create a map of empty test cluster connections that use fakes
+	clusterConnections := GetManagedClusterConnections()
+	clusterConnection := clusterConnections["cluster1"]
+
+	// add some cluster roles through the fake client set on the test cluster
+	cr := v1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test",
+			Labels: map[string]string{
+				"label1": "foo",
+			},
+		},
+	}
+	_, err := clusterConnection.KubeClient.RbacV1().ClusterRoles().Create(context.TODO(), &cr, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("got an error trying to a create cluster role: %v", err)
+	}
+
+	cr = v1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test2",
+		},
+	}
+	_, err = clusterConnection.KubeClient.RbacV1().ClusterRoles().Create(context.TODO(), &cr, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("got an error trying to create cluster role: %v", err)
+	}
+
+	// create a simpleClusterRoleLister using the fake client set from the test cluster
+	l := simpleClusterRoleLister{
+		clusterConnection.KubeClient,
+	}
+
+	// list all the cluster roles through the lister
```

Comment:
That's why we wrote this set of simple listers that work directly with the client sets (fakes for these tests) instead of a cache.Indexer.  This test is testing the simpleClusterRoleLister which is set on the test managed cluster connection with the fake client set.  

See testdata.go ...
```
// Get a managed cluster connection that uses fake client sets
func getManagedClusterConnection(clusterName string) *util.ManagedClusterConnection {
	// create a ManagedClusterConnection that uses client set fakes
	clusterConnection := &util.ManagedClusterConnection{
		KubeClient:                  fake.NewSimpleClientset(),
		KubeExtClientSet:            apiextensionsclient.NewSimpleClientset(),
		VerrazzanoOperatorClientSet: clientset.NewSimpleClientset(),
		DomainClientSet:             domclientset.NewSimpleClientset(),
		CohClusterClientSet:         cohcluclientset.NewSimpleClientset(),
		IstioClientSet:              istioClientsetFake.NewSimpleClientset(),
		IstioAuthClientSet:          istioAuthClientset.NewSimpleClientset(),
	}
...
	clusterConnection.ClusterRoleLister = &simpleClusterRoleLister{
		kubeClient: clusterConnection.KubeClient,
	}
```

and fakes.go ...
```
// simple ClusterRoleLister implementation
type simpleClusterRoleLister struct {
	kubeClient kubernetes.Interface
}

// List lists all the ClusterRoles.
func (s *simpleClusterRoleLister) List(selector labels.Selector) (ret []*rbacv1.ClusterRole, err error) {
	var clusterRoles []*rbacv1.ClusterRole

	list, err := s.kubeClient.RbacV1().ClusterRoles().List(context.TODO(), metav1.ListOptions{})
	if err != nil {
		return nil, err
	}
	for i := range list.Items {
		clusterRole := list.Items[i]
		if selector.Matches(labels.Set(clusterRole.Labels)) {
			clusterRoles = append(clusterRoles, &clusterRole)
		}
	}
	return clusterRoles, nil
}

// Get retrieves the ClusterRole for a given name.
func (s *simpleClusterRoleLister) Get(name string) (*rbacv1.ClusterRole, error) {
	return s.kubeClient.RbacV1().ClusterRoles().Get(context.TODO(), name, metav1.GetOptions{})
}

// InvokeHTTPHandler sets up a HTTP handler
func InvokeHTTPHandler(request *http.Request, path string, handler func(http.ResponseWriter, *http.Request)) *httptest.ResponseRecorder {
	responseRecorder := httptest.NewRecorder()
	router := mux.NewRouter()
	router.HandleFunc(path, handler)
	router.ServeHTTP(responseRecorder, request)

	return responseRecorder
}

```

Reply:
ok, i understand.  but it makes me question if this is something that can be reasonably tested in a unit test - maybe it needs to be an integration test instead.... again, no need to do anything here